systems/subscriber/registry/pkg/db/subscriber_repo_test.go

```diff
@@ -56,60 +57,399 @@ func (u UkamaDbMock) ExecuteInTransaction2(dbOperation func(tx *gorm.DB) *gorm.D
 	return nil
 }
 
-func TestSubscriber_Add(t *testing.T) {
+func setupTestDB(t *testing.T) (*sql.DB, sqlmock.Sqlmock, int_db.SubscriberRepo) {
 	db, mock, err := sqlmock.New()
 	if err != nil {
 		t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
 	}
-	defer db.Close()
+
 	gdb, _ := gorm.Open(postgres.New(postgres.Config{
 		DSN:                  "sqlmock_db_0",
 		DriverName:           "postgres",
 		Conn:                 db,
 		PreferSimpleProtocol: true,
 	}), &gorm.Config{})
+
 	repo := int_db.NewSubscriberRepo(&UkamaDbMock{
 		GormDb: gdb,
 	})
-	dateStr := "07-03-2023"
 
-	subscriber := int_db.Subscriber{
+	return db, mock, repo
```

Comment:
It is good to have a TestDB setup function, but here we have a problem, which is we are returning the sqlmock DB (wrapping the real sql.DB) that we absolutely do not need for testing because we not interacting directly the the Real DB in our whole code base. Instead we interacting with the DB through gorm, so  for the testing also, we are only interested in the gorm DB. So we should return the gorm Db (which wraps the sqlmock.Db, wich itself wraps the sql.Db) instead of returning the sqlmock.DB
so either do  `return gdb, mock` if you manually wants to create the repo, or `return mock, repo` if you want the repo creation being taking care of inside the setup function.